### PR TITLE
Validate objects/instances are registered in Send Request.

### PR DESCRIPTION
**Pull request by sbernard31**
This aims to implements : #1472

### CODE REVIEW

**Comment by sbernard31** [comment]
@JaroslawLegierski any opinion about this feature ? do you want to review it ?

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/send/SendHandler.java [hunk 1]
```diff
@@ -44,10 +47,18 @@ public void removeListener(SendListener listener) {
...
+        final SendResponse sendResponse = validateSendRequest(registration, request);
+        SendableResponse<SendResponse> response = new SendableResponse<>(sendResponse, new Runnable() {
             @Override
             public void run() {
-                fireDataReceived(registration, request.getTimestampedNodes(), request);
+                if (sendResponse.isSuccess()) {
+                    fireDataReceived(registration, request.getTimestampedNodes(), request);
+                } else {
+                    onError(registration, String.format("Invalid Send Request, server returns %s %s", //
+                            sendResponse.getCode().getName(), //
```
Comment (JaroslawLegierski) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/send/SendHandler.java:58]: also as above "//"

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/send/SendHandler.java [hunk 2]
```diff
@@ -44,10 +47,18 @@ public void removeListener(SendListener listener) {
...
-        SendableResponse<SendResponse> response = new SendableResponse<>(SendResponse.success(), new Runnable() {
+        final SendResponse sendResponse = validateSendRequest(registration, request);
+        SendableResponse<SendResponse> response = new SendableResponse<>(sendResponse, new Runnable() {
             @Override
             public void run() {
-                fireDataReceived(registration, request.getTimestampedNodes(), request);
+                if (sendResponse.isSuccess()) {
+                    fireDataReceived(registration, request.getTimestampedNodes(), request);
+                } else {
+                    onError(registration, String.format("Invalid Send Request, server returns %s %s", //
```
[the remark this answers is not in the document]
Comment (sbernard31) [reply leshan-server-core/src/main/java/org/eclipse/leshan/server/send/SendHandler.java:57]: Nope, this is to force formatter to return to next line. :slightly_smiling_face: 
Sometime I use this formatter trick to have code formatted in a way which seems better to me.

**Comment by sbernard31** [comment]
@JaroslawLegierski any news about it ? I would like to know if I can integrate it in `master` ?

**Comment by JaroslawLegierski** [comment]
> @JaroslawLegierski any news about it ? I would like to know if I can integrate it in `master` ?

Yes, of course - it's a very good idea to add this feature to the 'master' branch

**Pull request merged**